components/match_ticker/commons/match_ticker_display_components_new.lua

```diff
@@ -0,0 +1,285 @@
+---@field match table
+local Details = Class.new(
+	function(self, args)
+		assert(args.match, 'No Match passed to MatchTickerDetails class')
+		self.root = mw.html.create('div'):addClass('match-details')
+		self.hideTournament = args.hideTournament
+		self.onlyHighlightOnValue = args.onlyHighlightOnValue
+		self.match = args.match
+	end
+)
+
+---@return Html
+function Details:create()
+	local highlightCondition = HighlightConditions.match or HighlightConditions.tournament
+	if highlightCondition(self.match, {onlyHighlightOnValue = self.onlyHighlightOnValue}) then
+		self.root:addClass(HIGHLIGHT_CLASS)
+	end
+
+	return self.root
+		:node(self:streams())
+		:node(self:tournament())
+		:node(self:countdown())
+end
+
+---It will display both countdown and date of the match so the user can select which one to show
+---@return Html
+function Details:countdown()
+	local match = self.match
+
+	local dateString
+	if Logic.readBool(match.dateexact) then
+		local timestamp = DateExt.readTimestamp(match.date) + (Timezone.getOffset(match.extradata.timezoneid) or 0)
+		dateString = DateExt.formatTimestamp('F j, Y - H:i', timestamp) .. ' '
+				.. (Timezone.getTimezoneString(match.extradata.timezoneid) or (Timezone.getTimezoneString('UTC')))
+	else
+		dateString = mw.getContentLanguage():formatDate('F j, Y', match.date) .. (Timezone.getTimezoneString('UTC'))
+	end
+
+	local countdownArgs = {
+		date = dateString,
+		finished = match.finished
+	}
+
+	local countdownDisplay = mw.html.create('span')
+		:addClass('match-countdown')
+		:node(Countdown._create(countdownArgs))
+
+	return countdownDisplay
+end
+
+---@return Html?
+function Details:streams()
+	local match = self.match
+	local links = mw.html.create('div')
+		:addClass('match-streams')
+
+	mw.logObject(match.stream, 'match.stream')
+
+	if Table.isNotEmpty(match.stream) then
+		local streams = {}
+
+		-- Standardize the stream data to always use the platform as key (because of the new format ex: twitch_en_2)
+		for rawHost, stream in pairs(match.stream) do
+			local streamParts = mw.text.split(rawHost, '_', true)
+			if #streamParts == 3 then
+				local key = StreamLinks.StreamKey(rawHost)
+				streams[key.platform] = stream
+			else
+				streams[rawHost] = stream
+			end
+		end
+
+		local streamLinks = ''
+
+		-- Iterate over the streams and create the different links
+		for platformName, targetStream in pairs(streams) do
+			local streamLink = mw.ext.StreamPage.resolve_stream(platformName, targetStream)
+
+			if streamLink then
+				local icon = '<i class="lp-icon lp-icon-21 lp-' .. (PLATFORM_TO_ICON[platformName] or platformName) .. '"></i>'
+
+				-- TL.net specific
+				if platformName == 'stream' then
+					streamLinks = streamLinks ..
+						Page.makeExternalLink(icon, 'https://tl.net/video/streams/' .. streamLink)
+				else
+					streamLinks = streamLinks ..
+						Page.makeInternalLink({}, icon, 'Special:Stream/' .. platformName .. '/' .. streamLink)
+				end
+			end
+		end
```

Comment:
Use the standard library Steam Links (from #4372) for it all

Reply:
Copy pasted since the last time we talked you mentioned that we can refactor this once the related PR is merged. Should I wait for the PR then?

Reply:
Oh, I think you misunderstood what I mean then. I meant that as long as "nothing breaks" by merging one or the other first, not having a blocking dependency between the two PRs is fine. 
Since the non-compatible parts of this PR is locked behind a feature flag (effectively - needs a parameter to be set), there's no blocking dependency

Reply:
So just to be 100% aligned, it could be used like this since we have the feature flag?:
```suggestion
	if Table.isNotEmpty(match.stream) then
		links:wikitext(StreamLinks.display(StreamLinks.filterStreams(match.stream)))
	end
```

Reply:
Yes